lib/tasks/destroy_unused_providers.rb

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env ruby
+
+# Remove unused issuers
+#
+# When the IdP gets deployed, it generates a list of issuers that need to be removed and emails them
+# to us. These emails are available in Zendesk - check the "Suspended Tickets" bucket for the most
+# recent ticket titled:
+# "[production] identity-idp error: ServiceProviderSeeder::ExtraServiceProviderError".
+#
+# This script iterates over a list of those issuers, outputs the data that is to be deleted, and
+# then requires user confirmation before deleting the issuer and associated models.
+
+class DestroyUnusedProviders
+  attr_reader :destroy_list
+  def initialize(unused_issuers)
+    @destroy_list = unused_issuers.map { |issuer| DestroyableRecords.new(issuer) }
+  end
+
+  def run
+    @destroy_list.each do |records|
+      records.print_data
+
+      Rails.logger.debug do
+        "Type 'yes' and hit enter to continue and " \
+                         "destroy this service provider and associated records:\n"
+      end
+      continue = gets.chomp
+
+      if continue != 'yes'
+        Rails.logger.debug 'You have indicated there is an issue. Aborting script'
+        break
+      end
+
+      records.destroy_records
+    end
+    nil
+  end
+
+  class DestroyableRecords
+    attr_reader :sp, :int, :issuer
+
+    def initialize(issuer)
+      @issuer = issuer
+      @sp = ServiceProvider.includes(:in_person_enrollments).find_by(issuer: issuer)
+      @int = Agreements::Integration.includes(
+        :partner_account,
+        iaa_orders: [:iaa_gtc],
+      ).find_by(issuer: issuer)
+    end
+
+    def integration_usages
+      int.integration_usages
+    end
+
+    def iaa_orders
+      int.iaa_orders
+    end
+
+    def in_person_enrollments
+      sp.in_person_enrollments
+    end
+
+    def iaa_gtc
+    end
+
+    def print_data
+      Rails.logger.debug { "You are about to delete a service provider with issuer #{sp.issuer}" }
+      Rails.logger.debug { "The partner is #{int.partner_account.name}" }
+      Rails.logger.debug "\n\n"
+
+      Rails.logger.debug 'Attributes:'
+      Rails.logger.debug sp.attributes
+      Rails.logger.debug "\n"
+
+      Rails.logger.debug '********'
+      Rails.logger.debug 'Integration:'
+      Rails.logger.debug int.attributes
+      Rails.logger.debug "\n"
```

Comment:
WDYT of logging this in YAML format? in case we ever have to copy back to a `.yml` file?
```suggestion
      Rails.logger.debug 'Attributes:'
      Rails.logger.debug sp.attributes.to_yaml
      Rails.logger.debug "\n"

      Rails.logger.debug '********'
      Rails.logger.debug 'Integration:'
      Rails.logger.debug int.attributes.to_yaml
      Rails.logger.debug "\n"
```

Reply:
sounds good to me!

Reply:
added in [`91d0b32` (#8905)](https://github.com/18F/identity-idp/pull/8905/commits/91d0b3244f1ca136deb8dfa3755d9d5c32f9e657)